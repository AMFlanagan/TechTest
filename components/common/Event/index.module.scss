.event {
  background-color: #222222;
  margin: 16px 0;
  border-radius: 5px;
  overflow: hidden;
}

.eventHeader {
  display: flex;
  align-items: flex-end;
  padding: 8px;
  height: 40px;
}
