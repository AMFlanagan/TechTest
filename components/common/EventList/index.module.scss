.list {
  width: 100%;
  max-width: 800px;
  margin: 0 auto;
}

.title {
  padding: 8px 0;
  color: green;
}

.eventsContainer {
  max-width: 800px;
  margin: 0 auto;
}
