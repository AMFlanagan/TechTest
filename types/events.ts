export enum Sport {
  FOOTBALL = 'football',
  RUGBY = 'rugby',
  HORSE_RACING = 'horse-racing',
}

export enum FootballMarket {
  RESULT = 'Result',
  NEXT_GOAL = 'Next goal',
}

export enum RugbyMarket {
  RESULT = 'Result',
  NEXT_TRY = 'Next Try',
}

export enum HorseRacingMarket {
  RESULT = 'Result',
}

export interface IOutcome {
  id: number
  name: string
  odds: number
}

export type Market = FootballMarket | HorseRacingMarket | RugbyMarket

export interface IMarket {
  type: Market
  outcomes: IOutcome[]
}

export interface IEvent {
  id: number
  sport: Sport
  eventName: string
  markets: IMarket[]
  startTime: string
}
