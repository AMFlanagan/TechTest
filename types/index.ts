export interface IEvent {
  marketId: number
  outcomes: string[]
}
