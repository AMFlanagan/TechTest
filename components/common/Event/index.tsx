import React from 'react'
import { IEvent } from 'types'
import Market from '../Market'

import styles from './index.module.scss'

interface IProps {
  event: IEvent
}

const Event: React.FC<IProps> = (props: IProps) => {
  return (
    <div className={styles.event}>
      <div className={styles.eventHeader}>{props.event.eventName}</div>
      <Market
        eventName={props.event.eventName}
        eventId={props.event.id}
        market={props.event.markets[0]}
      />
    </div>
  )
}

Event.displayName = 'Event'

export default Event
