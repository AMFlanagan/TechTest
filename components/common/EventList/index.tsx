import React from 'react'
import { IEvent } from '../../../types'

import Event from '../Event'

import styles from './index.module.scss'

interface IProps {
  events: IEvent[]
}

const EventList: React.FC<IProps> = (props: IProps) => {
  return (
    <div className={styles.list}>
      <div className={styles.eventsContainer}>
        {props.events.map((event: IEvent, i: number) => {
          return <Event key={i} event={event} />
        })}
      </div>
    </div>
  )
}

EventList.displayName = 'EventList'

export default EventList
