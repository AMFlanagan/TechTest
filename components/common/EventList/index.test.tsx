import EventList from '.'
import { render } from '@testing-library/react'
import { Sport } from '../../../types'
import Event from '../Event'

jest.mock('../Event', () => jest.fn(() => <div>Event</div>))

describe('EventList', () => {
  it('renders as expect', () => {
    const events = [
      {
        id: 1,
        eventName: 'Event Name',
        markets: [],
        sport: Sport.FOOTBALL,
        startTime: '01/02/04',
      },
    ]

    const { asFragment, getByText } = render(
      <EventList events={events} title={'Football'} />
    )

    expect(getByText('Event')).toBeTruthy()
    expect(getByText('Football')).toBeTruthy()
    expect(Event).toBeCalledWith(
      expect.objectContaining({
        event: events[0],
      }),
      expect.anything()
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
