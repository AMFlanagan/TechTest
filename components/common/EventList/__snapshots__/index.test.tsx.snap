// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventList renders as expect 1`] = `
<DocumentFragment>
  <div
    class="list"
  >
    <div
      class="title"
    >
      Football
    </div>
    <div
      class="eventsContainer"
    >
      <div>
        Event
      </div>
    </div>
  </div>
</DocumentFragment>
`;
